core/src/main/java/org/elasticsearch/cluster/routing/allocation/RebalanceDecision.java

```diff
@@ -19,45 +19,87 @@
 
 package org.elasticsearch.cluster.routing.allocation;
 
+import org.elasticsearch.cluster.node.DiscoveryNode;
 import org.elasticsearch.cluster.routing.allocation.decider.Decision;
 import org.elasticsearch.cluster.routing.allocation.decider.Decision.Type;
 import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
 
+import java.io.IOException;
 import java.util.Collections;
 import java.util.Map;
 
+import static org.elasticsearch.cluster.routing.allocation.AllocateUnassignedDecision.nodeDecisionsToXContent;
+
 /**
  * Represents a decision to move a started shard to form a more optimally balanced cluster.
  */
 public final class RebalanceDecision extends RelocationDecision {
     /** a constant representing no decision taken */
-    public static final RebalanceDecision NOT_TAKEN = new RebalanceDecision(null, null, null, null, null, Float.POSITIVE_INFINITY);
+    public static final RebalanceDecision NOT_TAKEN = new RebalanceDecision(null, null, null, null, Float.POSITIVE_INFINITY, false);
 
     @Nullable
     private final Decision canRebalanceDecision;
     @Nullable
     private final Map<String, NodeRebalanceResult> nodeDecisions;
-    private float currentWeight;
-
-    public RebalanceDecision(Decision canRebalanceDecision, Type finalDecision, String finalExplanation) {
-        this(canRebalanceDecision, finalDecision, finalExplanation, null, null, Float.POSITIVE_INFINITY);
-    }
+    private final float currentWeight;
```

Comment:
that's true, its not needed anymore as the node results no longer maintain their weights as a comparison point (just using weight ranking now) so i'll remove this